perf pipeline: Switch to node 14.x

radical opened the pull request:
- Switch to node 14.x instead of 18.x, because the latter doesn't seem
to be compatible with ubuntu 18.x:
`node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)`

- remove the existing `nodejs` package before installing a new
one, because these machines are shared, and would get affected by
installations performed by other runs.

- Fix the case where pre-requisites would fail to install, but then helix would exit with exit code 0, and it would "look" as if everything went fine. This was happening because we set `_commandExitCode=1`, but helix sets this itself based on `$?`. So, to fix this we exit with `(exit 1)`, and helix should be able to pick that up, thus causing the whole job to fail, as expected

- Install node, and jsvu only for the wasm jobs, as they are not needed by others

radical committed:
perf pipeline: Switch to node 14.x

- Switch to node 14.x instead of 18.x, because the latter doesn't seem
to be compatible with ubuntu 18.x:
`node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)`

- Also, remove the existing `nodejs` package before installing a new
one, because these machines are shared, and would get affected by
installations performed by other runs.

radical commented:
/azp run runtime-wasm-perf

radical committed:
perf pipeline: install wasm pre-reqs only on wasm jobs

radical committed:
perf pipeline: correctly return exit code, so helix can pick that up and set _commandExitCode

radical commented:
/azp run runtime-wasm-perf

DrewScoggins commented on eng/pipelines/coreclr/templates/run-performance-job.yml:
```diff
@@ -79,10 +86,7 @@ jobs:
-        sudo apt-get -y install nodejs &&
-        npm install --prefix $HELIX_WORKITEM_PAYLOAD jsvu -g &&
-        $HELIX_WORKITEM_PAYLOAD/bin/jsvu --os=linux64 --engines=v8,javascriptcore &&
+        $(HelixPreCommandsWasmOnLinux) &&
```
Does this evaluate to blank or to the string if the variable is empty? I can't remember. Secondly, wouldn't we want to include the `&&` in the create of `HelixPreCommandsWasmOnLinux` so that we don't end up with an extra `&&` lying around when we don't want it.

radical replied:
Yeah, the blank thing might be a problem because the variable name is also under the condition. I'll fix that.

re:&&, I'll set the value to `echo` for non-wasm cases. Adding `&&` at the end of the command sounds wrong because it might, or might not be valid where it is being used

radical committed:
Set a default value for non-wasm cases

radical commented:
/azp run runtime-wasm-perf

radical committed:
Change the prereqs condition so a failure needs to be explicitly marked

This allows cases which don't care about this, to continue working,
because they won't set `$PERF_PREREQS_INSTALL_FAILED`, and default to
continuing with the job.

radical commented:
/azp run runtime-wasm-perf

radical committed:
Use --user with pip install

radical commented:
/azp run runtime-wasm-perf

radical commented:
@DrewScoggins I have a run from this in dotnet-runtime-perf - 20220817.1 . Does that look good? There are some failures on the wasm jobs, but those are related to recent ILLinker changes. I'll fix those.
The other jobs seem to be running too, but I checked only with some random sampling.

DrewScoggins commented on eng/testing/performance/microbenchmarks.proj:
```diff
@@ -148,10 +148,10 @@
-          $(WorkItemCommand) --bdn-artifacts $(ArtifactsDirectory) --bdn-arguments="--anyCategories $(BDNCategories) $(ExtraBenchmarkDotNetArguments) $(CoreRunArgument)";
-        else
+        if [ "x$PERF_PREREQS_INSTALL_FAILED" = "x1" ]; then
           echo "\n\n** Error: Failed to install prerequisites **\n\n"; export _commandExitCode=1;
```
This line should also exit 1, not use the export command

radical replied:
good catch!

DrewScoggins commented:
Just the one other comment, other than that LGTM.

radical commented on eng/testing/performance/microbenchmarks.proj:
```diff
@@ -148,10 +148,10 @@
-          $(WorkItemCommand) --bdn-artifacts $(ArtifactsDirectory) --bdn-arguments="--anyCategories $(BDNCategories) $(ExtraBenchmarkDotNetArguments) $(CoreRunArgument)";
-        else
+        if [ "x$PERF_PREREQS_INSTALL_FAILED" = "x1" ]; then
           echo "\n\n** Error: Failed to install prerequisites **\n\n"; export _commandExitCode=1;
```
```suggestion
          echo "\n\n** Error: Failed to install prerequisites **\n\n"; (exit 1);
```

radical committed:
Update eng/testing/performance/microbenchmarks.proj

radical commented:
Opened https://github.com/dotnet/runtime/issues/74087, and https://github.com/dotnet/performance/issues/2562 for the new failures.

radical commented:
And https://github.com/dotnet/runtime/issues/74088 .

Any objections to merging this right away to unblock the pipeline, since the changes are strictly in performance only files?

radical commented:
/backport to release/7.0-rc1
